artemis-cli/src/test/java/org/apache/activemq/cli/test/CliProducerTest.java

```diff
@@ -88,6 +90,124 @@ public void testSendMessage() throws Exception {
       checkSentMessages(session, address, null);
    }
 
+   @Test
+   public void testBooleanMessageProperties() throws Exception {
```

Comment:
can't you make a single test testMessageProperties... have multiple messages (or maybe just one) and validate them all in a single test?

Reply:
Yes. I could combine all the independent property tests into a single test. However, it's not clear to me what benefit that would provide.

Reply:
I'd guess that Clebert was just thinking about it starting X brokers and making X connections, sessions, producers currently to test something carried on the message alone that could still be done with 1 of each.

I tend to prefer more granular tests. Although, I'd personally go further and think the CLI bits could probably have been unit tested without any broker etc since its just a case of verifying the arg is decoded as expected, and ensuring things are set as expected on the message before it is sent...the clients sending of a message with props is covered elsewhere already)

Reply:
I consolidated all the individual tests into one.

Reply:
I like the idea of unit tests vs. integration tests so I implemented that.

Reply:
I actually attempted to use a mock to begin with but wasn't familiar enough (or patient enough) to get it working. Thanks for the commit. I've squashed and pushed.

Reply:
I actually meant push to main since I already reviewed them, should have been more specific given the context hehe, woops. Merged now :)